frontend/src/libs/components/checkbox/checkbox.tsx

```diff
@@ -1,17 +1,38 @@
+import { useFormController } from '#libs/hooks/hooks.js';
+import {
+  type FormControl,
+  type FormFieldPath,
+  type FormFieldValues,
+} from '#libs/types/types.js';
+
 import styles from './styles.module.scss';
 
-type Properties = {
+type Properties<T extends FormFieldValues> = {
+  control: FormControl<T, null>;
   label: string;
+  name: FormFieldPath<T>;
   onChange: () => void;
+  isChecked?: boolean;
 };
 
-const Checkbox: React.FC<Properties> = ({ label, onChange }) => {
+const Checkbox = <T extends FormFieldValues>({
+  control,
+  label,
+  name,
+  onChange,
+  isChecked,
+}: Properties<T>): JSX.Element => {
+  const { field } = useFormController({ name, control });
+
   return (
     <label>
       <input
+        {...field}
+        name={name}
         className={styles['checkbox']}
         type="checkbox"
         onChange={onChange}
+        checked={isChecked}
```

Comment:
we should pass field.value instead of isChecked here

Reply:
done